crates/iceberg-catalog/src/catalog/namespace.rs

```diff
@@ -312,3 +338,160 @@ pub(crate) fn validate_namespace_ident(namespace: &NamespaceIdent) -> Result<()>
 
     Ok(())
 }
+
+fn set_namespace_location_property(
+    namespace_props: &mut NamespaceProperties,
+    warehouse: &GetWarehouseResponse,
+    namespace_id: NamespaceIdentUuid,
+) -> Result<()> {
+    let mut location = namespace_props.get_location();
+
+    // NS locations should always have a trailing slash
+    location.as_mut().map(Location::with_trailing_slash);
+
+    // For customer specified location, we need to check if we can write to the location.
+    // If no location is specified, we use our default location.
+    let location = if let Some(location) = location {
+        if warehouse.storage_profile.is_allowed_location(&location) {
+            location
+        } else {
+            return Err(ErrorModel::bad_request(
+                "Namespace location is not a valid sublocation of the storage profile",
+                "NamespaceLocationForbidden",
+                None,
+            )
+            .into());
+        }
+    } else {
+        warehouse
+            .storage_profile
+            .default_namespace_location(namespace_id)?
+    };
+
+    namespace_props.insert(&location);
+    Ok(())
+}
+
+fn update_namespace_properties(
+    previous_properties: Option<HashMap<String, String>>,
+    updates: NamespaceProperties,
+    removals: Option<Vec<String>>,
+) -> (HashMap<String, String>, UpdateNamespacePropertiesResponse) {
+    let mut properties = previous_properties.unwrap_or_default();
+
+    let mut changes_updated = vec![];
+    let mut changes_removed = vec![];
+    let mut changes_missing = vec![];
+
+    if let Some(removals) = removals {
+        for key in removals {
```

Comment:
`for key in removals.unwrap_or_default().into_iter()` ?

Reply:
fixed